design/open/commit-metadata/airflow.md

```diff
@@ -0,0 +1,175 @@
+
+### Scope
+
+We add to each commit metadata for these items:
+
+* DAG run.  The URL of the DAG run in Airflow that generated this commit.
+  This is the URL of [Get a DAG run][get-dag-run] in the REST API.
+* Optionally, additional DAG identifiers such as:
+  * Git commit digest
+  * Airflow system identifier
+  * Airflow logical ("execution") date
+  * Notes (an Airflow metadata concept)
+  * Conf.
+
+  We allow all items that can be retrieved by [Get a DAG run][get-dag-run]
+  in the REST API, and copy terminology from there.
+
+  All optionals must be exactly that.  Most importantly, conf can contain
+  secrets and _must_ be optional.
+  
+### Naming
+
+We define a simple naming scheme that allows some future-compatibility.  All
+items use the prefix "`::lakefs::Airflow::`", staking a claim to the
+"`::lakefs::`" prefix on commit metadata.
+
+lakeFS supports only string metadata.  All plain suffixes are assumed to be
+plain strings.  If a suffix ends in `[...]`, that encodes a _type_.  We
+initially define the types that are needed to integrate Airflow.
+
+We add these suffixes to encode how to access the DAG run:
+
+* `url[url:id]`: The DAG run
+* `url[url:ui]`: URL to access the Airflow UI for the run.
+
+The REST API suggests the following suffixes:
+
+* `dag_run_id`: string
+* `dag_id`: string
+* `logical_date[iso8601]`: ISO 8601
+
+  `execution_date` is deprecated by Airflow and equal to `logical_date`, so
+  it is _not_ included.
+* `start_date[iso8601]`: ISO 8601
+* `end_date[iso8601]`: ISO 8601
+* `data_interval_start[iso8601]`: ISO 8601
+* `data_interval_end[iso8601]`: ISO 8601
+* `last_scheduling_decision[iso8601]`:  ISO 8601
+* `run_type`: string
+* `state`: string (presumably always "success")
+* `external_trigger[boolean]`: boolean encoded as "`true`" or "`false`"
+* `conf[json]`: string encoding a JSON object of conf
+* `note`: string
+
+#### Changing Airflows
+
+If the domain name of the Airflow system changes then URLs stop working.
+The DAG run URL is still a useful _identifier_, but it cannot be used to
+query for information.
+
+In a second phase, we may offer a simple future-proof method to allow
+changing URLs: Configure a number of "URL endpoints" on lakeFS.  They will
+look like variable expansions ("`http://$my_airflow/path/to/dag`") in commit
+metadata.  Configuring the current Airflow base URL in the lakeFS
+configuration allows both uniqueness and forwards compatibility to work.
+
+Users are not required to use these shortcuts, of course -- and small or
+test installations might choose to avoid them entirely.
+
+## lakeFS UI
+
+Integration of the Airflow UI into the lakeFS UI.  The UI will allow viewing
+the associated metadata from the UI.  When viewing a commit, the Airflow UI
+URL will become a link.  We shall also open the DAG in a frame above the
+commit metadata if this is available.
+
+### Flow
+
+When displaying a commit, and if enabled, scan its metadata for URLs of type
```

Comment:
So the underlying commit metadata in lakeFS will only contain a URL/identifier? Then the rest of the metadata is queried upon displaying it?

Reply:
No.  We put a whole lot of metadata on there, everything in "Scope".  But the only thing that adds a link is a url:ui.